Editor/AGS.CScript.Compiler/Preprocessor.cs

```diff
@@ -158,19 +170,32 @@ private string GetNextWord(ref string text)
 		private string GetNextWord(ref string text, bool trimText, bool includeDots)
 		{
 			int i = 0;
-			while ((i < text.Length) && 
-				   (text[i].IsScriptWordChar() ||
-				    (includeDots && (text[i] == '.')))
-				   )
+			while (i < text.Length)
 			{
+				if (char.IsWhiteSpace(text[i]) || text[i] == ';' || text[i] == ',' || text[i] == '(' || text[i] == ')' || (!includeDots && text[i] == '.'))

```

Comment:
I don't think that's a fully correct inversion of the previous condition: this does not include other symbols that may be met in script, such as `+`, `-`, `*`, `/`, square brackets, figure brackets, and anything else I forgot about.

Would not it be possible to keep the previous condition, but also detect invalid chars right away?

Reply:
It looks like the following change would be enough, if added right after the original loop:

```
            if (i < text.Length && text[i] > 127)
            {
                RecordError(ErrorCode.InvalidCharacter, "Invalid character detected in script at position " + i.ToString() + " in '" + text + "'");
                string res = text;
                text = string.Empty; // need to end line
                return res;
            }
```

Reply:
Thank you! I think that would work, yes. I won't be back to my computer until a few days (using my phone), if you want to do this change directly feel free.

If you want to modify this branch too with this change I think you can do it either as a moderator of ags or alternatively I added you as a collaborator in my fork.

At a later time I plan to contribute some additional (more complete) tests.

Reply:
I found time and implemented your change and it works fine!